### PR TITLE
Add Support CtrlP simply.

### PR DESCRIPTION
**Pull request by tsuyoshicho**
Setting see README diff

This PR add CtrlP Plugin with spelunker.

Bad spell listup and select to jump first target word.

Check context and operation. 1 or ALL, fix or ignore add.

### CODE REVIEW

**Comment by kamykn** [comment]
Hi, @tsuyoshicho !

Good job! The collaboration between spelunker.vim and CtrlP is an interesting collaboration!

Just let me ask one question. I cannot execute the `:CtrlPSPell` command with my vim8.
Is there any problem in my vim?

```
Plug 'tsuyoshicho/spelunker.vim' " @feature/ctrlp branch            
let g:enable_spelunker_vim_on_readonly = 0                
let g:spelunker_white_list_for_user = ['kamykn', 'vimrc'] 
let g:spelunker_check_type = 2                            
set spelllang=en,cjk                                      
let g:spelunker_highlight_type = 1                        
                                                          
Plug 'ctrlpvim/ctrlp.vim'                                 
let g:ctrlp_map = '<c-p>'                                 
let g:ctrlp_cmd = 'CtrlP'                                 
let g:ctrlp_working_path_mode = 'ra'                      
                                                          
" ctrlp ext                                               
let g:ctrlp_extensions = get(g:, 'ctrlp_extensions', [])  
      \ + ['spelunker']                                   
```

One thing can be considered, this command should be move to `plugin/spelunker.vim`. 🤔

**Comment by tsuyoshicho** [comment]
I understand it, please wait a while

**Comment by tsuyoshicho** [comment]
https://github.com/junegunn/vim-plug/blob/897ce5e2fad66bdb39dc94894f16dde89eee87f7/doc/plug.txt#L166-L167

vim-plug branch setting need below?:

```
Plug 'tsuyoshicho/spelunker.vim', { 'branch': 'feature/ctrlp" }
```

And then, `<c-p>` or `:CtrlPSpell` (miss spelled sorry) re-try.

(Spell miss fix and push)

File: README.md [hunk 1]
```diff
@@ -226,6 +226,23 @@ Zt
...
+
+Need setting, see below:
+
+```vim
+" ctrlp ext
+let g:ctrlp_extensions = get(g:, 'ctrlp_extensions', [])
+      \ + ['spelunker']
+```
+
+Start `:CtrlPSPell` then list up bad spell.
```
[the remark this answers is not in the document]
Comment (tsuyoshicho) [reply README.md]: fix

**Comment by tsuyoshicho** [comment]
> If one thing can be considered, this line should be move to plugin/spelunker.vim. 🤔
>> command! CtrlPSpell :call ctrlp#init(ctrlp#spelunker#id())

There is no problem with the change.
It just mimics the other bundled plug-ins to avoid inconvenience.

**Comment by tsuyoshicho** [comment]
Success test code

`> vim -u test.vim -N` (vimrc test.vim, nocompatible)

1. vim start above command line
2. insert test string with badspell (like "testcode")
3. and any CR
   ```
   testcode


   ```
4. `:CtrlPSpell` exec
5. show `testcode`
6. select and jump OK.

```test.vim:vim
let s:vim_plug_url='https://github.com/junegunn/vim-plug'
if !filereadable(expand('~/.vim/vim-plug/plug.vim'))
    call system("git clone " . s:vim_plug_url . " " . $HOME . "/.vim/vim-plug/")
endif
source ~/.vim/vim-plug/plug.vim
call plug#begin('~/.vim/plugged')

Plug 'tsuyoshicho/spelunker.vim', { 'branch': 'feature/ctrlp' }
Plug 'ctrlpvim/ctrlp.vim'

call plug#end()

let g:enable_spelunker_vim_on_readonly = 0
let g:spelunker_white_list_for_user = ['kamykn', 'vimrc']
let g:spelunker_check_type = 2
set spelllang=en,cjk
let g:spelunker_highlight_type = 1

let g:ctrlp_map = '<c-p>'
let g:ctrlp_cmd = 'CtrlP'
let g:ctrlp_working_path_mode = 'ra'

" ctrlp ext
let g:ctrlp_extensions = get(g:, 'ctrlp_extensions', []) + ['spelunker']
```

**Comment by kamykn** [comment]
> I cannot execute the :CtrlPSPell command with my vim8.

> `:CtrlPSpell `

Ahh, I found it was misspelled. 🙈
This command is working well on my vim8.

> test.vim

This is helpful information for me.
I'm going to try adding it to unit test!

If there are no more changes, I will merge this PR.

**Comment by tsuyoshicho** [comment]
No more Change it! :-)

Please merge 👍

**Pull request merged**